src/dialogs/permissions.tsx

```diff
@@ -110,23 +117,29 @@ function mode_to_args(mode) {
     return chmod_args.join(",");
 }
 
-const EditPermissionsModal = ({ dialogResult, selected, path }) => {
+const EditPermissionsModal = ({ dialogResult, selected, path } : {
+    dialogResult: DialogResult<void>,
+    selected: FolderFileInfo | null,
+    path: string,
+}) => {
     const { cwdInfo } = useFilesContext();
+    let selectedIsCwd = false;
 
     // Nothing selected means we act on the current working directory
     if (!selected) {
         const directory_name = basename(path);
-        selected = { ...cwdInfo, isCwd: true, name: directory_name };
+        selected = { ...cwdInfo, name: directory_name, category: null, to: null };
+        selectedIsCwd = true;
     }
 
     const [owner, setOwner] = useState(selected.user);
-    const [mode, setMode] = useState(selected.mode);
+    const [mode, setMode] = useState(selected.mode ?? 0);
     const [group, setGroup] = useState(selected.group);
-    const [errorMessage, setErrorMessage] = useState(undefined);
-    const [accounts, setAccounts] = useState(null);
-    const [groups, setGroups] = useState(null);
+    const [errorMessage, setErrorMessage] = useState<string | null>(null);
+    const [accounts, setAccounts] = useState<PasswdUserInfo[] | null>(null);
+    const [groups, setGroups] = useState<EtcGroupInfo[] | null>(null);
     const [isExecutable, setIsExecutable] = useState((mode & 0b001001001) === 0b001001001);
-    const [selinuxContext, setSELinuxContext] = useState(null);
+    const [selinuxContext, setSELinuxContext] = useState("");
```

Comment:
This seems wrong, we compare `!== null` when rendering? so this should be the same as above `string | null`

Reply:
oh yeah, my bad